fe/fe-core/src/main/java/org/apache/doris/analysis/StringLiteral.java

```diff
@@ -135,6 +135,9 @@ public boolean isMinValue() {
 
     @Override
     public String toSqlImpl() {
+        if (value.equals("\\")) {
+            value = "\\\\";
+        }
         return "'" + value.replaceAll("'", "''") + "'";
```

Comment:
maybe we need a util to process string escape.
mysql doc: https://dev.mysql.com/doc/refman/8.4/en/string-literals.html